packages/@lwc/engine-core/src/framework/hydration.ts

```diff
@@ -423,12 +425,15 @@ function validateClassAttr(vnode: VBaseElement, elm: Element, renderer: Renderer
     // are rendered during SSR. This needs to be accounted for when validating.
     if (scopedToken) {
         if (!isUndefined(className)) {
-            className = `${scopedToken} ${className}`;
+            className = `${scopedToken} ${className} ${stylesheetTokenHost}`.trim();
```

Comment:
Why the extra `trim()` here? Is it safe to assume whitespace is trimmed in all cases (e.g. static vs non-static nodes)?

Reply:
My thought was that if `stylesheetTokenHost` isn't present then `className` would have an extra space at the end of it.

So the final output would look like "x-main_main slds-hello ".

But now that you mention it, I'm not sure if `scopedToken` or `className` are guaranteed to have spaces already trimmed.

It may be a better idea to conditionally check if `stylesheetToken` exists and append it if it does instead.